commit ac0d9eaec501dcefe50f6bf6d6fa16fbb8509fe1
Author: Noralf Trønnes <noralf@tronnes.org>
Date:   Thu Feb 23 14:29:56 2017 +0100

    drm/tinydrm: mipi-dbi: Fix field width specifier warning
    
    This warning is seen on 64-bit builds in functions:
       'mipi_dbi_typec1_command':
       'mipi_dbi_typec3_command_read':
       'mipi_dbi_typec3_command':
    
    >> drivers/gpu/drm/tinydrm/mipi-dbi.c:65:20: warning: field width specifier '*' expects argument of type 'int', but argument 5 has type 'size_t {aka long unsigned int}' [-Wformat=]
          DRM_DEBUG_DRIVER("cmd=%02x, par=%*ph\n", cmd, len, data); \
                           ^
       include/drm/drmP.h:228:40: note: in definition of macro 'DRM_DEBUG_DRIVER'
         drm_printk(KERN_DEBUG, DRM_UT_DRIVER, fmt, ##__VA_ARGS__)
                                               ^~~
    >> drivers/gpu/drm/tinydrm/mipi-dbi.c:671:2: note: in expansion of macro 'MIPI_DBI_DEBUG_COMMAND'
         MIPI_DBI_DEBUG_COMMAND(cmd, parameters, num);
         ^~~~~~~~~~~~~~~~~~~~~~
    
    Fix by casting 'len' to int in the macro MIPI_DBI_DEBUG_COMMAND().
    There is no chance of overflow.
    
    Signed-off-by: Noralf Trønnes <noralf@tronnes.org>
    Signed-off-by: Dave Airlie <airlied@redhat.com>

diff --git a/drivers/gpu/drm/tinydrm/mipi-dbi.c b/drivers/gpu/drm/tinydrm/mipi-dbi.c
index 3f7892eb8e67..186ffd9628d1 100644
--- a/drivers/gpu/drm/tinydrm/mipi-dbi.c
+++ b/drivers/gpu/drm/tinydrm/mipi-dbi.c
@@ -62,9 +62,9 @@
 	if (!len) \
 		DRM_DEBUG_DRIVER("cmd=%02x\n", cmd); \
 	else if (len <= 32) \
-		DRM_DEBUG_DRIVER("cmd=%02x, par=%*ph\n", cmd, len, data); \
+		DRM_DEBUG_DRIVER("cmd=%02x, par=%*ph\n", cmd, (int)len, data);\
 	else \
-		DRM_DEBUG_DRIVER("cmd=%02x, len=%zu\n", cmd, len); \
+		DRM_DEBUG_DRIVER("cmd=%02x, len=%zu\n", cmd, (int)len); \
 })
 
 static const u8 mipi_dbi_dcs_read_commands[] = {
